### PR TITLE
Spring reference readme

### PR DESCRIPTION
**Pull request by yiliuTo**
Introduction of sdk/spring/spring-reference.yaml

### CODE REVIEW

File: sdk/spring/SPRING_REFERENCE_DOC.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+- msdocs: Product documentation.
+- artifacts: Package list of a service feature.
+- artifactId: Artifact-id of each package.
+- groupId: Group-id of each package.
+- versionGA: The latest GA version if exists.
+- versionPreview: The latest preview version if exists.
+- type: To mark if the package is Spring library, Azure service client library or management library, supported values are `spring`, `client` and `mgmt`.
+- links: Link list of each package. For below link properties, we refer values in https://azure.github.io/azure-sdk/releases/latest/all/java.html.
+- repopath: Maven repository url.
+- msdocs: Documentation url of repository azure-docs-for-java.
```
Comment (chenrujun) [inline sdk/spring/SPRING_REFERENCE_DOC.md:32]: We already have `msdocs` in line 23.

How about writing these as comments in spring-reference.yml?

**Review by chenrujun — CHANGES_REQUESTED** [review]

File: sdk/spring/spring-reference.yml [hunk 2]
```diff
@@ -1,42 +1,55 @@
...
-      id: spring-boot-bom
-      description: BOM for Microsoft Azure Spring Boot support.
-      artifacts:
+# This file is aimed to build package reference of Azure Spring libraries to support an Intellij plugin for Azure SDK reference.
+#
+#The yaml follows structure of Spring Initializr's library mapping schema(https://github.com/spring-io/start.spring.io/blob/master/start-site/src/main/resources/application.yml) and thus is compatible of keys under property of `dependencies` to describe a library.
+#To make things clear, we call sdk/spring/spring-reference.yml as Spring yaml, and Spring Initializr's library mapping schema as template yaml.
+#For those keys not listed in Spring yaml, we can add them in case of future usage.
+#
+#Hierarchy of Spring yaml is organized as : Service -> Feature -> Package. Taking Cosmos DB starter as example, it is described as Cosmos DB -> Spring Data Cosmos -> azure-spring-boot-starter-cosmos.
```
Comment (chenrujun) [inline sdk/spring/spring-reference.yml:7]: nit: Better to have s space after `#`.

**Review by backwind1233 — APPROVED** [review]
Cool commets

File: sdk/spring/spring-reference.yml [hunk 3]
```diff
@@ -1,42 +1,55 @@
...
+
+- name: Azure Spring Boot Bom                                                                       # Name of Azure service which the spring library relies on.
+                                                                                                    #    - For Spring library of which the underlying package is covered by [java-packages.csv](https://github.com/Azure/azure-sdk/blob/master/_data/releases/latest/java-packages.csv), we use its underlying library's `ServiceName`as the service name.
+                                                                                                    #    - For the rest like AAD starter, we create the service name ourselves, and try to follow the specification of java-pacakges.csv.
```
Comment (chenrujun) [inline sdk/spring/spring-reference.yml:11]: typo: pacakges

File: sdk/spring/spring-reference.yml [hunk 4]
```diff
@@ -1,42 +1,55 @@
...
+  content:                                                                                          # Features under an Azure Service.
+    - name: Azure Spring Boot Bom                                                                   # Name of a feature under the associated Azure service which the spring library relies on.
+                                                                                                    #    - For Spring library of which the underlying package is covered by [java-packages.csv](https://github.com/Azure/azure-sdk/blob/master/_data/releases/latest/java-packages.csv), we use its underlying library's `DisplayName`as the feature name.
+                                                                                                    #    - For the rest, we create the feature name ourselves, and try to follow the specification of java-pacakges.csv.
```
Comment (chenrujun) [inline sdk/spring/spring-reference.yml:15]: Same here.

File: sdk/spring/spring-reference.yml [hunk 5]
```diff
@@ -1,42 +1,55 @@
...
+      id: spring-boot-bom                                                                           # Id of feature name that is not used now and reserved in case of future usage for reference.
```
Comment (chenrujun) [inline sdk/spring/spring-reference.yml:16]: `Id of feature name that is not used ...` -> `Id of feature name. It is not used ...`

File: sdk/spring/spring-reference.yml [hunk 6]
```diff
@@ -1,42 +1,55 @@
...
+            repopath: https://search.maven.org/artifact/com.azure.spring/azure-spring-boot-bom/3.2.0/jar/               # Maven repository url.
+            github: https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/boms/azure-spring-boot-bom              # Github repository url.
+          springProperties:                                                                         # All spring specific properties copied from the template yaml.
+            starter: false                                                                          # Boolean value to mark if the library is a starter or not.
+            compatibilityRange: "[2.2.11.RELEASE,2.4.0-M1)"                                         # The compatible range for Spring Boot version.
+            mappings:                                                                               # If a library shows different compatibilites towards different Spring Boot/Cloud version, we use this property for their version mapping.
```
Comment (chenrujun) [inline sdk/spring/spring-reference.yml:30]: Typo: compatibilites.

We can use Intellij to check typos.

**Review by chenrujun — CHANGES_REQUESTED** [review]

**Review by chenrujun — APPROVED** [review]
LGTM.

**Pull request merged**